src/app/parcels/add/AddParcelForm.tsx

```diff
@@ -0,0 +1,195 @@
+        return new Date(
+            date.getFullYear(),
+            date.getMonth(),
+            date.getDate(),
+            time.getHours(),
+            time.getMinutes()
+        );
+    };
+
+    const submitForm = async (): Promise<void> => {
+        setSubmitDisabled(true);
+        let inputError;
+        if (fields.shippingMethod === "Collection") {
+            inputError = checkErrorOnSubmit(formErrors, setFormErrors);
+        } else {
+            inputError = checkErrorOnSubmit(formErrors, setFormErrors, [
+                "voucherNumber",
+                "packingDate",
+                "timeOfDay",
+                "shippingMethod",
+            ]);
+        }
+        if (inputError) {
+            setSubmitError(Errors.submit);
+            setSubmitDisabled(false);
+            return;
+        }
+        if (fields.packingDate === null || fields.timeOfDay === null) {
+            setSubmitDisabled(false);
+            return;
+        }
+
+        const packingDateTime = mergeDateAndTime(
+            new Date(fields.packingDate),
+            new Date(fields.timeOfDay)
+        );
+
+        let collectionDateTime = new Date();
+        if (fields.shippingMethod === "Collection") {
+            if (fields.collectionDate === null || fields.collectionTime === null) {
+                setSubmitDisabled(false);
+                return;
+            }
+
+            collectionDateTime = mergeDateAndTime(
+                new Date(fields.collectionDate),
+                new Date(fields.collectionTime)
+            );
+        }
+
+        const isDelivery = fields.shippingMethod === "Delivery";
+
+        const formToAdd = {
+            client_id: props.id,
+            packing_datetime: packingDateTime.toISOString(),
+            voucher_number: fields.voucherNumber,
+            collection_centre: isDelivery ? "Delivery" : fields.collectionCentre,
+            collection_datetime: isDelivery ? null : collectionDateTime.toISOString(),
+        };
+        try {
+            await insertParcel(formToAdd);
+            router.push("/clients/");
+        } catch (error: unknown) {
+            if (error instanceof Error) {
+                setSubmitError(Errors.external);
+                setSubmitErrorMessage(error.message);
+                setSubmitDisabled(false);
+            }
+        }
+        setSubmitDisabled(false);
+    };
+    return (
+        <CenterComponent>
+            <StyledForm>
+                <FormHeading>Add Parcel</FormHeading>
+                {formSections.map((Card, index) => {
+                    return (
+                        <Card
+                            key={index}
+                            formErrors={formErrors}
+                            errorSetter={errorSetter}
+                            fieldSetter={fieldSetter}
+                            fields={fields}
+                        />
+                    );
+                })}
+                <CenterComponent>
```

Comment:
Do we need this CenterComponent?I tried removing it and it doesn't seem to be doing anything.

Reply:
Another interesting observation is the button changes colour when I add/remove the CenterComponent. Dont know why that is the case.